resources/projects/plugins/robot_windows/generic/generic.c

```diff
@@ -11,22 +11,29 @@ void wb_robot_window_init() {
 }
 
 void wb_robot_window_step(int time_step) {
-  const char *message = wb_robot_wwi_receive_text();
-  if (message) {
-    if (!wbu_generic_robot_window_handle_messages(message)) {
-      // JavaScript -> C protocol description:
-      //   [deviceName:commandTag[=commadState][,]]*
-      // example:
-      //   "e-puck:forward,ds0:enable,myMotor0:value=1.2"
-      char *tokens = strdup(message);
-      char *token = NULL;
-      while ((token = wbu_string_strsep(&tokens, ","))) {
-        char *command = strdup(token);
-        char *first_word = wbu_string_strsep(&command, ":");
-        if (!wbu_generic_robot_window_parse_device_control_command(first_word, command))
-          wbu_generic_robot_window_parse_device_command(first_word, command);
+  int length = 0;
+  const char *message = wb_robot_wwi_receive(&length);
+  int character_read = 0;
+  while (character_read < length) {
+    if (message) {
+      if (!wbu_generic_robot_window_handle_messages(message)) {
+        // JavaScript -> C protocol description:
+        //   [deviceName:commandTag[=commadState][,]]*
+        // example:
+        //   "e-puck:forward,ds0:enable,myMotor0:value=1.2"
+        char *tokens = strdup(message);
+        char *token = NULL;
+        while ((token = wbu_string_strsep(&tokens, ","))) {
+          char *command = strdup(token);
+          char *first_word = wbu_string_strsep(&command, ":");
+          if (!wbu_generic_robot_window_parse_device_control_command(first_word, command))
+            wbu_generic_robot_window_parse_device_command(first_word, command);
+        }
       }
     }
+    const int current_message_length = strlen(message) + 1;
+    character_read += current_message_length;
+    message += current_message_length;
```

Comment:
That looks wrong if message == NULL:
```suggestion
      }
      const int current_message_length = strlen(message) + 1;
      character_read += current_message_length;
      message += current_message_length;
    }
```